### PR TITLE
[MetricsAdvisor] Added tests for the AnomalyAlertConfiguration CRUD operations

### LINKED ISSUE #15916: [MetricsAdvisor] Write tests for the AnomalyAlertConfiguration CRUD operations
Add tests for the service methods and related models.
- CreateAnomalyAlertConfiguration
- GetAnomalyAlertConfiguration
- GetAnomalyAlertConfigurations
- UpdateAnomalyAlertConfiguration
- DeleteAnomalyAlertConfiguration

### PR DESCRIPTION
**Pull request by kinelski**
Fixes https://github.com/azure/azure-sdk-for-net/issues/15916.

Tests kinda large because setup of complex models takes some lines of code.

### CODE REVIEW

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs [hunk 1]
```diff
@@ -0,0 +1,987 @@
...
+
+            await using var disposableConfig = await DisposableAlertConfiguration.CreateAlertConfigurationAsync(adminClient, configToCreate);
+
+            AnomalyAlertConfiguration createdConfig = await adminClient.GetAlertConfigurationAsync(disposableConfig.Id);
+
+            Assert.That(createdConfig.Id, Is.EqualTo(disposableConfig.Id));
+            Assert.That(createdConfig.Name, Is.EqualTo(configName));
+            Assert.That(createdConfig.Description, Is.Empty);
+            Assert.That(createdConfig.CrossMetricsOperator, Is.Null);
+            Assert.That(createdConfig.IdsOfHooksToAlert, Is.Not.Null.And.Empty);
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs:42]: what is the difference between having null and empty here? looks like the service will still succeed on creating a alertConfiguration even if no hooks are added.

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs [hunk 2]
```diff
@@ -0,0 +1,987 @@
...
+            Assert.That(updatedMetricAlertConfig1.AlertConditions.MetricBoundaryCondition, Is.Null);
+            Assert.That(updatedMetricAlertConfig1.AlertConditions.SeverityCondition, Is.Null);
+            Assert.That(updatedMetricAlertConfig1.AlertSnoozeCondition, Is.Null);
+            Assert.That(updatedMetricAlertConfig1.UseDetectionResultToFilterAnomalies, Is.True);
+        }
+
+        [RecordedTest]
+        public async Task UpdateAlertConfigurationWithEveryMemberAndGetInstance()
+        {
+            MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient();
```
Comment (maririos) [inline sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs:560]: I see what you mean with all the preparation for creating the tests :|

**Review by maririos — APPROVED** [review]

File: sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs (hunk 1 above)
Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs:42]: It should never be null. If no hooks are added, it's expected to be empty. You can see that I've added a `Is.Not.Null` check in every test, regardless of hooks being added or not.

Comment (maririos) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs:42]: Sorry, I commented in the wrong spot.
I mean for the input parameter. If the service allows no hooks, what is the difference between the user sending null and an empty list?

Comment (kinelski) [reply sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/AnomalyAlertConfigurationLiveTests.cs:42]: User can't send null. Currently they'll get an `ArgumentNullException` if they try to pass a null value to the `AnomalyAlertConfiguration` constructor.

Once [this design issue](https://github.com/Azure/azure-sdk-for-net/issues/16321) gets fixed, the empty list of hooks will be available (and not settable) by default, without being passed to the constructor. Possibly something like:
```cs
var config = new AnomalyAlertConfiguration();
config.IdsOfHooksToAlert.Add(hook);
```

**Pull request merged**